src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,152 @@
+        {
+            string? storedComplexValue = await defaultProperties.GetUnevaluatedPropertyValueAsync(PropertyInProjectFile);
+            ComplexTargetFramework values = BreakDownValue(storedComplexValue);
+            return await base.OnGetEvaluatedPropertyValueAsync(propertyName, ExtractValue(values, propertyName) ?? "", defaultProperties);
+        }
+
+        public override async Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            string? storedComplexValue = await defaultProperties.GetUnevaluatedPropertyValueAsync(PropertyInProjectFile);
+            ComplexTargetFramework values = BreakDownValue(storedComplexValue);
+            return await base.OnGetUnevaluatedPropertyValueAsync(propertyName, ExtractValue(values, propertyName) ?? "", defaultProperties);
+        }
+
+        //private static string? CommonCode(string propertyName, IProjectProperties defaultProperties)
+        //{
+        //    string? storedComplexValue = defaultProperties.GetUnevaluatedPropertyValueAsync(PropertyToBeSavedOnProjectFile); // TODO: change to TargetFrameworkProperty
+        //    ComplexTargetFramework values = BreakDownValue(storedComplexValue);
+        //    return ExtractValue(values, propertyName);
+        //}
+
+        /// <summary>
+        /// Extracts the specified value from the ComplexTargetFramework structure.
+        /// </summary>
+        /// <param name="values">The strucutre were all the values are.</param>
+        /// <param name="propertyName">The property's value we want to return.</param>
+        /// <returns></returns>
+        private static string? ExtractValue(ComplexTargetFramework values, string propertyName)
+        {
+            return propertyName switch
+            {
+                TargetFrameworkProperty => values.TargetFramework,
+                TargetOSProperty => values.Platform,
+                TargetOSVersionProperty => values.PlatformVersion,
+                _ => null,
+            };
+        }
+
+        /// <summary>
+        /// Combines the three values that make up the TargetFramework property.
+        /// I.e. net5.0, windows, 10.0 => net5.0-windows10.0
+        /// net5.0, null, null => net5.0
+        /// </summary>
+        /// <param name="targetFramework">The target framework value.</param>
+        /// <param name="targetPlatform">The target platform value.</param>
+        /// <param name="platformVersion">The platform version value.</param>
+        /// <returns></returns>
+        private static string ComputeValue(string targetFramework, string? targetPlatform, string? platformVersion)
+        {
+            if (targetPlatform != null) // TODO: Ensure TargetFramework is not null.
+            {
+                return targetFramework + "-" + targetPlatform + platformVersion;
+            }
+            return targetFramework;
+        }
```

Comment:
Could be written as:
```suggestion
        private static string ComputeValue(string targetFramework, string? targetPlatform, string? platformVersion) => targetPlatform != null ? $"{targetFramework}-{targetPlatform}{platformVersion}" : targetFramework;
```

Reply:
I'm not sure this is easier to read.

Reply:
Interesting. It is easier for me to read as it is a single expressions. IMO, single expressions are generally easier to read than separate statements; mainly due to the statements may/may not have relationships with each other. I generally like minimizing 'statement bloat' as it causes loss of intent. Honestly, I'd probably style this as 2 lines to make it more obvious.
```suggestion
        private static string ComputeValue(string targetFramework, string? targetPlatform, string? platformVersion) =>
            targetPlatform != null ? $"{targetFramework}-{targetPlatform}{platformVersion}" : targetFramework;
```

Anyway, readability is always preference as long as the logic is the same.

Reply:
At the end, the logic ended up being more complex. Thanks for the feedback, though.